src/linker/Linker/Driver.cs

```diff
@@ -780,29 +781,41 @@ public int Run (ILogger? customLogger = null, bool throwOnFatalLinkerException =
 
 			try {
 				p.Process (Context);
-			} catch (LinkerFatalErrorException lex) {
+			} catch (Exception e) when (FailCatch (e)) {
+				// Proces will be terminated in exception filter.
+				throw e;
+			} finally {
+				Context.FlushCachedWarnings ();
+				Context.Tracer.Finish ();
+			}
+
+			return Context.ErrorsCount > 0 ? 1 : 0;
+		}
+
+		/// <summary>
+		/// Handles exceptions in the linker. Prints error messages, then calls Environment.FailFast to create a dump with the full call stack.
+		/// </summary>
+		bool FailCatch (Exception e)
```

Comment:
Nit: I would call this something like `LogFatalFailure` or something like that. To note that it's solely about logging more info.